File: Leetcode/cpp/sol1252.cpp
// https://leetcode.com/problems/cells-with-odd-values-in-a-matrix/

class Solution {
public:
    void print(vector<int> arr){
        for(auto i:arr){
            cout << i << " ";
        }
        cout << endl;
    }
    int oddCells(int m, int n, vector<vector<int>>& indices) {

        vector<int> row(m, 0);
        vector<int> col(n, 0);
        for (auto i:indices){
            row[i[0]]++;
            col[i[1]]++;
        }
        int result = 0;
        for(int i = 0; i < m; i++){
            for(int j = 0; j < n; j++){
                if ((row[i] + col[j]) % 2 != 0){
                    result++;
                }
            }
        }

        return result;
    }
};

File: Leetcode/python/sol1252.py
# https://leetcode.com/problems/cells-with-odd-values-in-a-matrix/description/
class Solution:
    def is_odd(self, num :int) -> bool:
        return num % 2 != 0

    def oddCells(self, m: int, n: int, indices: List[List[int]]) -> int:

        matrix = [[0 ] *n for i in range(m)]
        for row, col in indices:
            for i in range(len(matrix[row])):
                matrix[row][i] += 1
            for i in range(len(matrix)):
                matrix[i][col] += 1

        rslt = 0
        for row in matrix:
            for val in row:
                if self.is_odd(val):
                    rslt+=1

        return rslt
